### ISSUE TITLE
First pass at implementing a network management API

### ISSUE THREAD

**Issue by stgraber** [lxc/lxd#2349]
This is a tracking bug for work going around /1.0/networks.

We're looking at adding:
 - Ability to create new bridges managed by LXD
 - Configure them (IPv4, IPv6, tunneling, port security, backend driver, ...)
 - Add containers to them
 - Delete

I'm working on this in my "network" branch, adding all the API bits and all the new configuration keys.

This work will affect both the daemon (big API extension), client (support for "lxc network ..." commands) and "lxd init" (to setup an initial network).

**Comment by jamesbeedy** [comment]
Woohooo!! +1000

**Comment by cryptofuture** [comment]
Cool. I'm thinking about nat disabling option per container right now without creating extra bridges, for example for default lxdbr0 bridge.  
Seems like lxd-bridge iptables rule makes masquerade (POSTROUTING) for full lxd bridge subnet right now. And rejecting just forward per ip didn't work.  And I don't think that split POSTROUTING rules is a right way anywhere beside my test machine.  
Anyway, if creation of new bridges will be just easy as editing `/etc/default/lxd-bridge`...
Basically what I trying to say, it will be wonderful to see new improved network management for sure.
